test/src/unit-dense-reader.cc

```diff
@@ -0,0 +1,447 @@
+/**
+ * @file unit-dense-reader.cc
```

Comment:
Shouldn't this file be in `sm/query/readers/test`?

Reply:
These tests are using the C api so they cannot be moved to the proper unit test folder. It is testing the reader which is one of the hardest thing to test without using any APIs. We'll get there but for now we need to keep this here.